src/Data/List/Relation/Unary/Any/Properties.agda

```diff
@@ -624,6 +625,17 @@ module _ {P : B → Set p} where
     to∘from (y ∷ xs) f (there p) =
       P.cong there $ to∘from xs (f ∘ there) p
 
+------------------------------------------------------------------------
+-- reverse
+
+module _ {P : A → Set p} where
+  reverse⁺ : {xs : List A} → Any P xs → Any P (reverse xs)
```

Comment:
I would personally go via a

```agda
reverseAcc⁺ : (acc : List A) -> Any P xs -> Any P (reverseAcc acc xs)
```

proof. It is a straightforward induction on `Any P xs` and then `reverse⁺`
is a direct corollary.

It would also be nice to have

```agda
reverseAcc⁻ : Any P (reverseAcc acc xs) -> Any P acc ⊎ Any P xs
```

(This one is probably a pain to prove without using `Data.List.Properties`'s
`ʳ++-defn` and `++⁻`).

Reply:
> I would personally go via a
> 
> ```agda
> reverseAcc⁺ : (acc : List A) -> Any P xs -> Any P (reverseAcc acc xs)
> ```
> 
> proof. It is a straightforward induction on `Any P xs` and then `reverse⁺`
> is a direct corollary.

Would it be sensible to use the following?

```agda
reverseAcc⁺ : ∀ (xs : List A) (acc : List A) → Any P acc ⊎ Any P xs → Any P (reverseAcc acc xs)
```

I otherwise can't see how to deal with the base case in `reverseAcc⁺ : (acc : List A) -> Any P xs -> Any P (reverseAcc acc xs)`.

Reply:
Updated and squashed. I wonder if `reverseAcc⁻` can be more succinct?

Reply:
One thing you could do is not pattern-match on `xs` for `reverseAcc⁻`:
`Any P (reverseAcc acc xs)` becomes `Any P (reverse xs ++ acc)`
becomes `Any P (reverse xs) ⊎ Any P acc` and then if you already
have `reverse⁻` via `reverse⁺` & involution you're done.

But maybe it's more natural to do derive `reverse⁻` from `reverseAcc⁻`
like you did.

Reply:
I'm happy either way, either relying on `ʳ++-defn` and `++⁻` or on `reverseAcc⁺` and involution.